scipy/signal/tests/test_signaltools.py

```diff
@@ -483,6 +483,47 @@ def test_basic(self):
         win = signal.get_window(('kaiser', 8.0), 160)
         assert_raises(ValueError, signal.resample, sig, num, window=win)
 
+    def test_rfft(self):
+        # Make sure the speed up using rfft gives the same result as the normal 
+        # way using fft
+
+        x = np.linspace(0, 10, 20, endpoint=False)
+        y = np.cos(-x**2/6.0)
+        y_complex = y.copy()+complex(0,0)
+        # Upsample
+        assert_array_almost_equal(signal.resample(y, 100), signal.resample(y_complex, 100).real)
```

Comment:
Ok, so

1. It should match without this, so make sure there's not a bug that's causing it to mismatch.
2. If it should output real for real input, then the .real needs to be inside the resample function, not in the test.